tests/runner/test_ansible.py

```diff
@@ -38,10 +38,12 @@ def setUp(self):
                 'host_password': 'pass',
             })
         )
+        fd, self.host_file2 = tempfile.mkstemp()
```

Comment:
why need a fd here?

Reply:
could you rename it to a meaningful name?

Reply:
update to self.test_mpi_host_file, is it ok?